src/util/locking/FileSystemResourceLocker.ts

```diff
@@ -0,0 +1,146 @@
+import { createHash } from 'crypto';
+import { ensureDirSync, pathExists, readdir, rmdir } from 'fs-extra';
+import type { LockOptions, UnlockOptions } from 'proper-lockfile';
+import { lock, unlock } from 'proper-lockfile';
+import type { ResourceIdentifier } from '../../http/representation/ResourceIdentifier';
+import type { Finalizable } from '../../init/final/Finalizable';
+import { getLoggerFor } from '../../logging/LogUtil';
+import { createErrorMessage } from '../errors/ErrorUtil';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { AttemptSettings } from '../LockUtils';
+import { toRetryOnError, retryFunctionUntil } from '../LockUtils';
+import { joinFilePath } from '../PathUtil';
+import type { ResourceLocker } from './ResourceLocker';
+
+const defaultLockOptions: LockOptions = {
+  /** Resolve symlinks using realpath, defaults to true (note that if true, the file must exist previously) */
```

Comment:
I don't fully understand what is meant with "the file must exist previously"

Reply:
Since normally this is mapped to the file system, the path to the file must exists for a lock to be taken. If `realPath == true` then the path to the requested file on which to lock, will be resolved, thus must exist. Since I am mapping all paths to the same lockfolder, the realPath option is set to false, so as to not resolve any directories within the library.

Reply:
Perhaps instead of having a line giving the generic explanation of how the field works, instead have a comment here and below explaining why we set it to false? Are there things that would break if someone sets this to `true`? If yes this should not be a default option but a forced option in the constructor instead.